doc/parallel.md

```diff
@@ -0,0 +1,67 @@
+Parallel testing
+=========================
+Standard RapidCheck stateful testing can be used to find bugs that occur as a result of single threaded interaction with the system under test. However, another very common class of bugs are race conditions which by definition only appear when more than one thread is involved. RapidCheck can provoke race conditions in stateful tests by executing a subset of the commands in a command sequence in parallel. A parallel test is conseridered successful if there is a possible interleaving of the commands in the test that can explain the result of the execution.
+
+## The anatomy of a parallel test ##
+The execution of a parallel test case can be divided into four phases: command generation, pre-condition verification, execution and verification. 
+
+During command generation three distinct command sequences are generated: the prefix, the left branch and the right branch. The prefix will be executed first, followed by the left and right branches that are executed in parallel. The exact order in which the commands in the left and right branches will be executed is not known to RapidCheck. This makes pre-condition verification more strict than in the sequential case as pre-condition verification for each test case has to be done for all possible command interleavings. The test case is discarded if any failing pre-condition is found.
```

Comment:
"is not known to RapidCheck"

Maybe add "... since it does not control the operating system scheduler."

I'd like to rephrase:
"This makes pre-condition verification more strict..."
...into..
"This requires pre-condition verification to be more strict..."

The strictness doesn't happen automatically, but it wouldn't work without it :)